CMakeLists.txt

```diff
@@ -481,7 +481,7 @@ export(PACKAGE SMTK)
 #
 # Required to happen before we include the docs directory
 set(MOAB_ROOT_DIR "${MOAB_ROOT_DIR}")
-set(SMTK_MODULE_DIR "${SMTK_SOURCE_DIR}/CMake")
+set(SMTK_MODULE_DIR "${CMAKE_BINARY_DIR}/lib/cmake/SMTK")
```

Comment:
I don't understand what `SMTK_MODULE_DIR` is used for. From the name, I would have guessed it gets added to the CMake module search path. Instead, maybe it's where the SMTK targets file lives in the build dir? Also, shouldn't we prefer `${CMAKE_BINARY_DIR}/lib/cmake/SMTK/${SMTK_VERSION}` so that multiple versions can coexist?